crates/parser/src/config/mod.rs

```diff
@@ -566,12 +596,40 @@ pub enum ConfigError {
     InvalidErrorThreshold(u64),
 }
 
+/// A `schemars::visit::Visitor` that sets a `default` for all values that are `const`. Also
+/// normalizes any schemas with `enum: ["singleValue"]` to `const: "singleValue"`, which is needed
+/// by the estuary UI in order to render properly.
+#[derive(Debug, Clone)]
+struct ConstDefaultSchemaVisitor;
+impl schemars::visit::Visitor for ConstDefaultSchemaVisitor {
+    fn visit_schema_object(&mut self, schema: &mut schemagen::SchemaObject) {
+        schemars::visit::visit_schema_object(self, schema);
+
+        if schema.enum_values.as_ref().map(|v| v.len()) == Some(1) {
+            let value = schema.enum_values.take().unwrap().pop().unwrap();
+            schema.const_value = Some(value);
```

Comment:
Do we need to `schema.enum_values = None` now?

Reply:
In this case, we do. According to the JSON schema spec `enum: ["foo"]` and `const: "foo"` are identical. But JSON Forms treats these two differently, and requires `const` to be used for a `discriminator`. This is just changing the representation from `enum` to `const`.

Reply:
Ahh, sorry, I misunderstood. Setting `enum_values` to `None` is handled by calling [`take()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.take) on the line above.